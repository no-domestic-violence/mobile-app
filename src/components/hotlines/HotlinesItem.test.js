import React from 'react';
import HotlinesItem from './index';
import { render, fireEvent } from '@testing-library/react-native';

describe('Hotline item component', () => {
  const mockHotlineItem = {
    _id: '5f9db611c7cc881787ba620b',
    city: 'Hamburg',
    country: 'Germany',
    organisation_name: 'Test name 2',
    phone: '+49 543 514 8358',
    website: 'www.help.de',
  };
  const mockMakeCall = jest.fn();
  it('renders hotline item from props', () => {
    const { queryByText, debug } = render(
      <HotlinesItem item={mockHotlineItem} />
    );
    expect(queryByText(mockHotlineItem.organisation_name)).not.toBeNull();
    expect(
      queryByText(`${mockHotlineItem.city}, tel:${mockHotlineItem.phone}`)
    ).not.toBeNull();
    expect(queryByText(`${mockHotlineItem.website}`)).not.toBeNull();
  });
  it('make a call on press with the righ number', () => {
    const { getByTestId } = render(
      <HotlinesItem item={mockHotlineItem} makeCall={mockMakeCall} />
    );
    fireEvent.press(getByTestId('makeCall'));
    expect(mockMakeCall).toHaveBeenCalledTimes(1);
    expect(mockMakeCall).toHaveBeenCalledWith(mockHotlineItem.phone);
  });
  it('should match snapshot', () => {
    const result = render(
      <HotlinesItem item={mockHotlineItem} makeCall={mockMakeCall} />
    ).toJSON();
    expect(result).toMatchSnapshot();
  });
});
