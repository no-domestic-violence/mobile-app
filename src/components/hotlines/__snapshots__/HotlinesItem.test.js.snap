// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotline item component should match snapshot 1`] = `
Array [
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FEF8E3",
        "borderRadius": 30,
        "flexDirection": "row",
        "flexShrink": 1,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flexShrink": 1,
          "flexWrap": "wrap",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexWrap": "wrap",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#242424",
              "flexShrink": 1,
              "fontWeight": "bold",
            }
          }
        >
          Test name 2
        </Text>
        <Text
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "flexShrink": 1,
              "flexWrap": "wrap",
            }
          }
        >
          Hamburg
          , tel:
          +49 543 514 8358
        </Text>
        <Text
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "flexShrink": 1,
              "flexWrap": "wrap",
            }
          }
        >
          website:
        </Text>
        <View>
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "blue",
              }
            }
            title="www.help.de"
            type="clear"
          >
            www.help.de
          </Text>
          <Text
            style={
              Object {
                "color": "#242424",
              }
            }
          >
             
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <
        color="#000"
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f095",
              "M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z",
            ],
            "iconName": "phone",
            "prefix": "fas",
          }
        }
        onPress={[Function]}
        size={25}
        testID="makeCall"
      />
    </View>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#cadeee",
        "height": 20,
      }
    }
  />,
]
`;
